lib/pry/pry_instance.rb

```diff
@@ -529,28 +529,29 @@ def should_print?
   # @return [String] The prompt.
   def select_prompt
     object = current_binding.eval('self')
-
     open_token = @indent.open_delimiters.any? ? @indent.open_delimiters.last :
       @indent.stack.last
 
-    c = Pry::Config.from_hash({
-                       :object         => object,
-                       :nesting_level  => binding_stack.size - 1,
-                       :open_token     => open_token,
-                       :session_line   => Pry.history.session_line_count + 1,
-                       :history_line   => Pry.history.history_line_count + 1,
-                       :expr_number    => input_array.count,
-                       :_pry_          => self,
-                       :binding_stack  => binding_stack,
-                       :input_array    => input_array,
-                       :eval_string    => @eval_string,
-                       :cont           => !@eval_string.empty?})
+    c = Pry::Config.new(nil).tap {|conf|
```

Comment:
It's rather pointless to pass `nil`. Would look simpler with `c = Pry::Config.new.tap {|conf|`

Reply:
The default parent is `Pry.config`, which we do not want here. 
When you want to explicitly opt for no parent, you explicitly pass `nil`. At least that's the current API and not worth changing in this PR imo.